### ISSUE TITLE
findOne and findByPk does not pass all the options to the hooks

### ISSUE THREAD

**Issue by maksimsemenov** [sequelize/sequelize#11190]
## What are you doing?

I'm trying to pass some app context to the `beforeFind` hook through options in `findByPk` and `findOne`:
```js
const entity = await model.findByPk(entityId, { context } )
// or
const entity = await model.findOne({ where: { id: entityId}, context } )
```
In both cases, in `beforeFind` hook I'm getting only some default options:
```js
beforeFind: async options => {
    console.log(options)
}
// in console I'm getting
{
       where: { [Symbol(and)]: [ [Object], [Object] ] },
       limit: null,
       plain: true,
       hooks: true,
       rejectOnEmpty: false
}
```
But when I pass the same prop to the find all, the context object is passing through:
```js
const entity = await model.findAll({ where: { id: entityId}, limit: 1, context } )
// and then in beforeFind hook, I see:
{
   context: {
       contextProp1: 123,
       ...
   },
   ... all other options
}
```
**To Reproduce**
1. Define models with `beforeFind` hook
2. Call `findByPk` or `findOne` and pass custom context object with options
```js
const entity = await model.findByPk(entityId, { context } )
// or
const entity = await model.findOne({ where: { id: entityId}, context } )
```
3. Check the `options` argument in `beforeFind` hook. The context property is missing.

From looking at the source code is not clear what is causing the bug. I couldn't find any code that will remove any custom property from options.

## What do you expect to happen?
I expect to see the custom context prop in the options object in `beforeFind` hook.
## What is actually happening?
The context prop is passed through only for `findAll` method, but not for `findByPk` or `findOne` (even if I pass `individualHooks: true`).

## Environment
Sequelize version: 5.9.4
Node Version: 10.15.3
OS: macOS Mojave
Tested with latest release:
- [x] No
- [ ] Yes, specify that version:

**Comment by SimonSchick** [comment]
I don't think we are making any guarantees about what non standard properties are passed on, therefor I wouldn't consider this a bug but rather a feature request.

**Comment by maksimsemenov** [comment]
@SimonSchick should I recreate the issue as a feature request?
This feature is very important for using Sequelize in serverless environment (AWS Lambda): in those environments we can rely on a global variable/state to inject some session data (like user id) into sequelize hooks, because it can be shared between multiple function invocations.

**Comment by papb** [comment]
No need to create a new issue.
